diff --git a/node_modules/@ethersproject/json-wallets/lib.esm/keystore.js b/node_modules/@ethersproject/json-wallets/lib.esm/keystore.js
index c0eee41..613f95d 100644
--- a/node_modules/@ethersproject/json-wallets/lib.esm/keystore.js
+++ b/node_modules/@ethersproject/json-wallets/lib.esm/keystore.js
@@ -9,7 +9,11 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
     });
 };
 import aes from "aes-js";
-import scrypt from "scrypt-js";
+import {scrypt as _syncScrypt, scryptAsync as _scrypt} from "@noble/hashes/scrypt";
+const scrypt = {
+    syncScrypt: (password, salt, N, r, p, dkLen, progressCallback) => _syncScrypt(password, salt, {N, r, p, dkLen, onProgress: progressCallback}),
+    scrypt: (password, salt, N, r, p, dkLen, progressCallback) => _scrypt(password, salt, {N, r, p, dkLen, onProgress: progressCallback})
+}
 import { getAddress } from "@ethersproject/address";
 import { arrayify, concat, hexlify } from "@ethersproject/bytes";
 import { defaultPath, entropyToMnemonic, HDNode, mnemonicToEntropy } from "@ethersproject/hdnode";
diff --git a/node_modules/@ethersproject/json-wallets/lib/keystore.js b/node_modules/@ethersproject/json-wallets/lib/keystore.js
index e981cf3..c35f178 100644
--- a/node_modules/@ethersproject/json-wallets/lib/keystore.js
+++ b/node_modules/@ethersproject/json-wallets/lib/keystore.js
@@ -56,7 +56,13 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.encrypt = exports.decrypt = exports.decryptSync = exports.KeystoreAccount = void 0;
 var aes_js_1 = __importDefault(require("aes-js"));
-var scrypt_js_1 = __importDefault(require("scrypt-js"));
+var _scrypt_js_1 = __importDefault(require("@noble/hashes/scrypt"));
+var scrypt_js_1 = {
+    default: {
+        syncScrypt: (password, salt, N, r, p, dkLen, progressCallback) => _scrypt_js_1.scrypt(password, salt, {N, r, p, dkLen, onProgress: progressCallback}),
+        scrypt: (password, salt, N, r, p, dkLen, progressCallback) => _scrypt_js_1.scryptAsync(password, salt, {N, r, p, dkLen, onProgress: progressCallback})
+    }
+}
 var address_1 = require("@ethersproject/address");
 var bytes_1 = require("@ethersproject/bytes");
 var hdnode_1 = require("@ethersproject/hdnode");
diff --git a/node_modules/@ethersproject/json-wallets/src.ts/keystore.ts b/node_modules/@ethersproject/json-wallets/src.ts/keystore.ts
index 938cb82..298e8fd 100644
--- a/node_modules/@ethersproject/json-wallets/src.ts/keystore.ts
+++ b/node_modules/@ethersproject/json-wallets/src.ts/keystore.ts
@@ -1,7 +1,11 @@
 "use strict";
 
 import aes from "aes-js";
-import scrypt from "scrypt-js";
+import {scrypt as _syncScrypt, scryptAsync as _scrypt} from "@noble/hashes/scrypt";
+const scrypt = {
+    syncScrypt: (password: Uint8Array, salt: Uint8Array, N: number, r: number, p: number, dkLen: number, progressCallback: ProgressCallback): Uint8Array => _syncScrypt(password, salt, {N, r, p, dkLen, onProgress: progressCallback}),
+    scrypt: (password: Uint8Array, salt: Uint8Array, N: number, r: number, p: number, dkLen: number, progressCallback: ProgressCallback): Promise<Uint8Array> => _scrypt(password, salt, {N, r, p, dkLen, onProgress: progressCallback})
+}
 
 import { ExternallyOwnedAccount } from "@ethersproject/abstract-signer";
 import { getAddress } from "@ethersproject/address";
