diff --git a/node_modules/@noble/hashes/esm/utils.js b/node_modules/@noble/hashes/esm/utils.js
index 946aec1..8a6580f 100644
--- a/node_modules/@noble/hashes/esm/utils.js
+++ b/node_modules/@noble/hashes/esm/utils.js
@@ -72,7 +72,7 @@ export function utf8ToBytes(str) {
     if (typeof str !== 'string') {
         throw new TypeError(`utf8ToBytes expected string, got ${typeof str}`);
     }
-    return new TextEncoder().encode(str);
+    return new Uint8Array(new TextEncoder().encode(str));
 }
 export function toBytes(data) {
     if (typeof data === 'string')
diff --git a/node_modules/@noble/hashes/src/utils.ts b/node_modules/@noble/hashes/src/utils.ts
index 8a8ecda..0c61540 100644
--- a/node_modules/@noble/hashes/src/utils.ts
+++ b/node_modules/@noble/hashes/src/utils.ts
@@ -88,7 +88,7 @@ export function utf8ToBytes(str: string): Uint8Array {
   if (typeof str !== 'string') {
     throw new TypeError(`utf8ToBytes expected string, got ${typeof str}`);
   }
-  return new TextEncoder().encode(str);
+  return new Uint8Array(new TextEncoder().encode(str));
 }
 
 export type Input = Uint8Array | string;
diff --git a/node_modules/@noble/hashes/utils.js b/node_modules/@noble/hashes/utils.js
index c0519ac..ee96b6f 100644
--- a/node_modules/@noble/hashes/utils.js
+++ b/node_modules/@noble/hashes/utils.js
@@ -83,7 +83,7 @@ function utf8ToBytes(str) {
     if (typeof str !== 'string') {
         throw new TypeError(`utf8ToBytes expected string, got ${typeof str}`);
     }
-    return new TextEncoder().encode(str);
+    return new Uint8Array(new TextEncoder().encode(str));
 }
 exports.utf8ToBytes = utf8ToBytes;
 function toBytes(data) {
